src/Corax/IndexEntry/IndexEntryWriter.cs

```diff
@@ -371,26 +371,79 @@ public void Write<TEnumerator>(int field, TEnumerator values, IndexEntryFieldTyp
         _dataIndex = dataLocation;
     }
 
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
     private int WriteNullsTableIfRequired<TEnumerator>(TEnumerator values, int dataLocation, ref IndexEntryFieldType indexEntryFieldLocation)
         where TEnumerator : IReadOnlySpanIndexer
+    {        
+        // If we have an small number of values, we just allocate the space for the nulls table in the stack. 
+        if (values.Length < 256)
+        {            
+            return WriteNullsTableIfRequiredStackalloc(values, dataLocation, ref indexEntryFieldLocation);
+        }
+
+        // If it is big enough, we will just allocate it on the heap and most likely will end up outside of the hot-path when
+        // the JIT compiler aggresively optimize it. 
+        return WriteNullsTableIfRequiredHeap(values, dataLocation, ref indexEntryFieldLocation);
+    }
+
+    private int WriteNullsTableIfRequiredHeap<TEnumerator>(TEnumerator values, int dataLocation, ref IndexEntryFieldType indexEntryFieldLocation)
+        where TEnumerator : IReadOnlySpanIndexer
     {
-        // We will include null values if there are nulls to be stored.           
         int nullBitStreamSize = values.Length / (sizeof(byte) * 8) + (values.Length % (sizeof(byte) * 8) == 0 ? 0 : 1);
-        byte* nullStream = stackalloc byte[nullBitStreamSize];
+        byte* nullBitStreamBuffer = stackalloc byte[nullBitStreamSize];
```

Comment:
I think you switched the method names. This is the `heap`, but allocates on the stack, and vice versa.

Can also always allocate 256 bytes, avoid the cost.

Reply:
And add a `Debug.Assert` to validate that the size isn't too big.

For the vast majority of cases, we'll get perfect code.